### PR TITLE
feat: Allow customizing ramdisk options

### PR DESCRIPTION
**Pull request by bebraw**
This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] tests
- [ ] documentation
- [ ] metadata

### Breaking Changes?

- [ ] yes
- [x] no

If yes, please describe the breakage.

### Please Describe Your Changes

This PR allows you to customize `ramdisk` options by letting the user pass an object containing the options to the `ramdisk` field.

I came by this requirement in a project where the development output doesn't fit the default ramdisk (it's a big build!).

### CODE REVIEW

File: lib/plugins/ramdisk.js [hunk 1]
```diff
@@ -75,7 +76,10 @@ module.exports = {
 
     this.log.info(`Ramdisk enabled`);
 
-    const plugin = new WebpackPluginRamdisk({ name: 'wps' });
+    const defaultOptions = { name: 'wps' };
+    const plugin = new WebpackPluginRamdisk(
+      isObject(options) ? { ...options, ...defaultOptions } : defaultOptions
```
Comment (TrySound) [inline lib/plugins/ramdisk.js:81]: Maybe better `typeof options === 'object'` without extra package? Arrays are not supported here anyway. Null will be spread without issues.

Comment (bebraw) [reply lib/plugins/ramdisk.js]: Sure, done.

**Comment by shellscape** [comment]
Oddly enough, I'm unable to get tests to run successfully locally.

**Comment by bebraw** [comment]
> Oddly enough, I'm unable to get tests to run successfully locally.

Yeah, I wonder what's up with that. The CI seems to pass.

**Comment by shellscape** [comment]
It's looking like a problem with puppeteer. Tried to reinstall it and ran into a wall:

```
Downloading Chromium r782078 - 128 Mb [====================] 100% 0.0s 
ERROR: Failed to set up Chromium r782078! Set "PUPPETEER_SKIP_DOWNLOAD" env variable to skip download.
{ [Error: EEXIST: file already exists, mkdir '/.../node_modules/puppeteer/.local-chromium/mac-782078']
```

**Comment by bebraw** [comment]
@shellscape That's an interesting one. I wonder if it's something temporary. I guess we could try to look into updating puppeteer though based on the error, it sounds like it had downloaded it already but failed to detect that.

**Comment by shellscape** [comment]
Turned out that it was due to a buggy version of Node 10. Used the latest stable of v10 and it worked. Still need to refactor these tests. After updating Ava we have a bunch of dangling handles.

**Comment by bebraw** [comment]
@shellscape Oh, wow. That's surprising.

**Comment by PlayMa256** [comment]
is that only me or we are the ones that find bugs on node versions? LOL

**Pull request merged**